Source/Fuse.Controls.Panels/Layouts/WrapLayout.uno

```diff
@@ -135,6 +153,8 @@ namespace Fuse.Layouts
 			var placements = new float4[elements.Count];
 			//minorMaxSize in each major row, assinged per element
 			var minorSizes = new float[elements.Count];
+			// save the available space for each row
+			var majorRest = new Dictionary<float, float>();
```

Comment:
That, or take the `Grid` approach which stores the Row/Column number on the properties of the elements. It's a bit more costly and not necessary here though I think. Your array idea is good.